addon/model.js

```diff
@@ -272,6 +272,15 @@ export default class MegamorphicModel extends EmberObject {
   }
 
   notifyPropertyChange(key) {
+    if (CUSTOM_MODEL_CLASS) {
+      // just super and move on for state flags
+      // this needs to match whatever we are notifying
+      // in our subscription to the notificationManager
+      if (['isNew', 'isDeleted'].indexOf(key) !== -1) {
+        super.notifyPropertyChange(key);
+        return;
+      }
```

Comment:
can we get a test for this please?

Reply:
fwiw something like 258 tests failed without this, and will fail again if anything new needs to be added to the list.